### PR TITLE
Fix for Xinerama alignment bugs

### PR DESCRIPTION
**Pull request by Francesco149**
This adds full support for xinerama, allowing the user to specify the desired xinerama head through the "xinerama_head" setting in the config file.
This fixes issue #172  where alignment would behave incorrectly on multiple xinerama heads with different height, as well as allowing for easier positioning on such a setup.
This can be toggled from BUILD_XINERAMA during the configure process.
Tested and working on gentoo x64 with two xinerama heads:
```
0	X: 1920	Y: 0	Width: 1080	Height: 1920
1	X: 0	Y: 0	Width: 1920	Height: 1080
```

Result:

.conkyrc (yes I am aware that this is the old syntax, I am too lazy to convert it right now):
```conkyrc
alignment bottom_left
xinerama_head 1 
background no
border_width 1
cpu_avg_samples 2
default_color white
default_outline_color white
default_shade_color white
draw_borders no
draw_graph_borders yes
draw_outline no
draw_shades no
use_xft yes
xftfont Hack:style=regular:size=9
gap_x 5
gap_y 5
minimum_size 5 5
net_avg_samples 2
no_buffers yes
out_to_console no
out_to_stderr no
extra_newline no
own_window yes
own_window_class Conky
own_window_type desktop
own_window_transparent yes
stippled_borders 0
update_interval 1.0
uppercase no
use_spacer none
show_graph_scale no
show_graph_range no
double_buffer yes

TEXT
$sysname $kernel $machine @ $nodename
$hr
${color grey}Uptime:$color $uptime
${color grey}Frequency (in MHz):$color $freq
${color grey}Frequency (in GHz):$color $freq_g
${color grey}RAM Usage:$color $mem/$memmax - $memperc%
${color grey}Swap Usage:$color $swap/$swapmax - $swapperc%
${color grey}CPU Usage:$color $cpu%
${color grey}Processes:$color $processes  ${color grey}Running:$color $running_processes
$hr
${color grey}/${goto 80}$color${fs_used /}/${fs_size /}${goto 215}${fs_bar 6 /}
${color grey}/home${goto 80}$color${fs_used /home}/${fs_size /home}${goto 215}${fs_bar 6 /home}
${color grey}~/wdtan${goto 80}$color${fs_used /home/loli/wdtan}/${fs_size /home/loli/wdtan}${goto 215}${fs_bar 6 /home/loli/wdtan}
${color grey}backup${goto 80}$color${fs_used /mnt/backup}/${fs_size /mnt/backup}${goto 215}${fs_bar 6 /mnt/backup}
$hr
${color grey}Up:$color ${upspeed enp4s0} ${color grey} - Down:$color ${downspeed enp4s0}
$hr
${color grey}Name              PID   CPU%   MEM%
${color lightgrey} ${top name 1} ${top pid 1} ${top cpu 1} ${top mem 1}
${color lightgrey} ${top name 2} ${top pid 2} ${top cpu 2} ${top mem 2}
${color lightgrey} ${top name 3} ${top pid 3} ${top cpu 3} ${top mem 3}
${color lightgrey} ${top name 4} ${top pid 4} ${top cpu 4} ${top mem 4}
```

### CODE REVIEW

File: extras/convert.lua [hunk 1]
```diff
@@ -46,7 +46,8 @@ local num_setting = {
     maximum_width = true, mpd_port = true, music_player_interval = true, net_avg_samples = true,
     own_window_argb_value = true, pad_percents = true, stippled_borders = true,
     text_buffer_size = true, top_name_width = true, total_run_times = true,
-    update_interval = true, update_interval_on_battery = true, xftalpha = true
+    update_interval = true, update_interval_on_battery = true, xftalpha = true, 
+	xinerama_head = true,
```
Comment (brndnmtthws) [inline extras/convert.lua]: Whitespace misaligned here.

**Comment by brndnmtthws** [comment]
Looks good to me, minus one nitpick. Can you also squash your changes into 1 commit?

**Comment by Francesco149** [comment]
first time I squash commits together, I hope I did it right. and yes that's indeed a typo marc

**Comment by Francesco149** [comment]
one moment, I'll also add and squash the typo and misaligned space fixes

**Comment by Francesco149** [comment]
I think that's it!

File: src/x11.cc [hunk 2]
```diff
@@ -449,6 +453,40 @@ static void update_workarea(void)
...
+	}
+
+	if (!XineramaIsActive(display)) {
+		return; /* has xinerama but isn't using it */
+	}
+
+	int heads = 0;
+	XineramaScreenInfo *si = XineramaQueryScreens(display, &heads);
+	if (!si) {
+		puts("warning: XineramaQueryScreen returned NULL, ignoring head settings");
```
Comment (brndnmtthws) [inline src/x11.cc]: Please use the macro `NORM_ERR()` here (from https://github.com/brndnmtthws/conky/blob/master/src/logging.h#L73-L77).

File: src/x11.cc [hunk 3]
```diff
@@ -449,6 +453,40 @@ static void update_workarea(void)
...
+		return; /* queryscreens failed? */
+	}
+
+	int i = head_index.get(*state);
+	if (i < 0 || i >= heads) {
+		puts("warning: invalid head index, ignoring head settings");
```
Comment (brndnmtthws) [inline src/x11.cc]: `NORM_ERR()` here too.

File: src/x11.cc [hunk 4]
```diff
@@ -449,6 +453,40 @@ static void update_workarea(void)
...
+		return;
+	}
+
+	XineramaScreenInfo *ps = &si[i];
+	workarea[0] = ps->x_org;
+	workarea[1] = ps->y_org;
+	workarea[2] = workarea[0] + ps->width;
+	workarea[3] = workarea[1] + ps->height;
+
+	printf("Fixed xinerama area to: %d %d %d %d\n",
```
Comment (brndnmtthws) [inline src/x11.cc]: Use `DBGP()` here.

**Comment by brndnmtthws** [comment]
Couple more small nits. After that, should be good to go.

**Pull request merged**